logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/asset/GetAssetSizeLimitUseCase.kt

```diff
@@ -0,0 +1,30 @@
+package com.wire.kalium.logic.feature.asset
+
+import com.wire.kalium.logic.feature.user.IsSelfATeamMemberUseCase
+
+fun interface GetAssetSizeLimitUseCase {
+    /**
+     * Returns the maximum size of an asset that can be uploaded to the Wire backend.
+     * @param isImage whether the asset to upload is an image or not
+     */
+    suspend operator fun invoke(isImage: Boolean): Long
+}
+
+internal class GetAssetSizeLimitUseCaseImpl internal constructor(
+    private val isSelfATeamMember: IsSelfATeamMemberUseCase
```

Comment:
I think we can inject here the `Dispatcher` and switching the execution here, so we don't have to do it on clients :D

Reply:
Not sure if we should switch the context on the use case level or rather on the repository, data layer level. Maybe something to discuss on next collective?